### PR TITLE
Public Works JSONify

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
Inspired by #19729 by @vickio, I decided to make a conversion of public works from `map::draw_map` to JSON, so here it is.

### CODE REVIEW

File: data/json/mapgen/public_works.json [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+      },
+      "toilets": {
+        "T": {}
+      },
+      "items": {
+        "P": { "item": "pool_table", "chance": 80 },
+        "e": { "item": "fridge", "chance": 80 }
+      },
+      "place_items": [
+        { "item": "hardware", "x": [ 2, 2 ], "y": [ 3, 8 ], "chance": 85 },
```
Comment (keyspace) [inline data/json/mapgen/public_works.json]: Could have just used `loot`, like [here](https://github.com/CleverRaven/Cataclysm-DDA/pull/20060/commits/1cbc4a6ee8fcff210ef3e5e034c1a157831d4588#diff-30040570d14f856a03c36dfc7f0e7c0eR57), e.g.:

``` json
"loot": {
    "r": { "item": "hardware", "chance": 85 }
}
```

No need to track coordinates.

**Comment by keyspace** [comment]
Will submit PR fixing above to to-merge branch.

File: data/json/mapgen/public_works.json (hunk 1 above)
Comment (ZhilkinSerg) [inline data/json/mapgen/public_works.json]: I've used `"items"` where possible without major impact (fridge 🍨 and pool table 🎱 ) . I'd have to change `"rows"` to make use of this in other item spawn locations, so I decided to skip this until all mapgen is jsonized and easily moddable.

**Comment by kolsurma** [comment]
#20296 mentions [this issue on the forum](http://smf.cataclysmdda.com/index.php?topic=12051.msg295688#msg295688) about the public works' ID changing.

The tests should probably iterate through the potential overmap tiles to at least check they exist.

**Pull request merged**